src/NServiceBus.Core/Recoverability/FirstLevelRetries/FirstLevelRetriesBehavior.cs

```diff
@@ -28,28 +29,26 @@ public FirstLevelRetriesBehavior(FlrStatusStorage storage, FirstLevelRetryPolicy
                 var messageId = context.Message.MessageId;
                 var pipelineUniqueMessageId = messageId;
 
-                var numberOfFailures = storage.GetFailuresForMessage(pipelineUniqueMessageId);
+                storage.MarkForFirstLevelRetry(pipelineUniqueMessageId, ExceptionDispatchInfo.Capture(ex));
```

Comment:
If we did we would have to change the SLR behavior because it assumes that storage holds number of failed processing attempts and not FLRRetries. I've change the name of the method to `AddFailedProcessingAttempt` to relfect what is actually stored.